cmd/pdatagen/internal/base_slices.go

```diff
@@ -232,7 +232,17 @@ func (es ${structName}) Resize(newLen int) {
 func (es ${structName}) AppendEmpty() ${elementName} {
 	*es.orig = append(*es.orig, &${originName}{})
 	return es.At(es.Len() - 1)
-} `
+}
+
+// Sort sorts the ${elementName} elements within ${structName} given the
+// provided less function so that two instances can be compared.
+//
+// Returns the same instance to allow nicer code like:
+//   assert.EqualValues(t, expected.Sort(lessFunc), actual.Sort(lessFunc))
+func (es ${structName}) Sort(less func(i, j int) bool) ${structName} {
```

Comment:
I want to see an example of how to write this function. Do we have everything else that is needed?

Reply:
It looks like this comment is misleading, and even the implementation could be improved. Please take a look at the latest commit (forgive the rebase, there was a conflict).

The improvement is to make the less function reusable by passing it two `pdata` elements, rather than two indices.

Improved example code:
```golang
		lessFunc := func(a, b pdata.Metric) bool {
			return a.Name() < b.Name()
		}
		assert.EqualValues(t, expected.Metrics().Sort(lessFunc), got.Metrics().Sort(lessFunc))
		...
		r.InstrumentationLibraryMetrics().Sort(func(a, b pdata.InstrumentationLibraryMetrics) bool {
			if a.InstrumentationLibrary().Name() == b.InstrumentationLibrary().Name() {
				return a.InstrumentationLibrary().Version() < b.InstrumentationLibrary().Version()
			}
			return a.InstrumentationLibrary().Name() < b.InstrumentationLibrary().Name()
		})
```

Reply:
@tigrannajaryan this is a bit different than golang, are we ok with this? Personally I like it.

Reply:
Yeah, I like it. Fairly standard sort func (even if not what golang's stdlib uses) and easy to understand.